app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,201 @@
+import com.codesoom.assignment.TaskRepositoryCleaner;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    abstract class Context_didCreateTwoTasks {
+        final int numberOfTasksCreated = 2;
+
+        @BeforeEach
+        void context() {
+            for (int i = 1; i <= numberOfTasksCreated; i++) {
+                final Task task = new Task(FIXTURE_TITLE + i);
+                service.createTask(task);
+            }
+        }
+    }
+
+    public static final String FIXTURE_TITLE = "title";
+    private TaskRepositoryCleaner repositoryCleaner;
+    private TaskService service;
+
+    @BeforeEach
+    void setup() {
+        final TaskRepository repository = new TaskRepository();
+        repositoryCleaner = new TaskRepositoryCleaner(repository);
+        service = new TaskService(repository);
+    }
+
+    @Nested
+    @DisplayName("getTaskList 메소드는")
+    class Describe_getTaskList {
+        @Nested
+        @DisplayName("생성되어 있는 할 일이 없다면")
+        class Context_didNotCreateTask {
+            @BeforeEach
+            void prepare() {
+                repositoryCleaner.deleteAllTasks();
+            }
+
+            @Test
+            @DisplayName("빈 목록을 반환한다")
+            void it_returnsEmptyList() {
+                List<Task> tasks = service.getTaskList();
+
+                assertThat(tasks).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("생성된 할 일들이 있다면")
+        class Context_didCreateTasks extends Context_didCreateTwoTasks {
+            @Test
+            @DisplayName("생성했던 할 일 목록을 반환한다")
+            void it_returnsTasks() {
+                List<Task> tasks = service.getTaskList();
+
+                assertThat(tasks).hasSize(numberOfTasksCreated);
+                final Task resultTask1 = tasks.get(0);
+                final Task resultTask2 = tasks.get(1);
+                assertThat(resultTask1.getId()).isEqualTo(1L);
+                assertThat(resultTask1.getTitle()).isEqualTo(FIXTURE_TITLE + 1);
+                assertThat(resultTask2.getId()).isEqualTo(2L);
+                assertThat(resultTask2.getTitle()).isEqualTo(FIXTURE_TITLE + 2);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("찾을 수 없는 id로 요청하면")
+        class Context_withNotFindableTaskId {
+            private final Long DELETED_TASK_ID = 1L;
+
+            @BeforeEach
+            void prepare() {
+                service.createTask(new Task(FIXTURE_TITLE));
+                service.deleteTask(DELETED_TASK_ID);
+            }
```

Comment:
좋습니다. 다만 이 한 줄은 없어도 문제없을 것 같군요.
```suggestion
            @BeforeEach
            void prepare() {
                //service.createTask(new Task(FIXTURE_TITLE));
                service.deleteTask(DELETED_TASK_ID);
            }
```